kitchen.vagrant.yml

```diff
@@ -0,0 +1,44 @@
+---
+driver:
+  name: vagrant
+  vm_hostname: salt
+  username: vagrant
+  cache_directory: false
+  customize:
+    usbxhci: 'off'
+  gui: false
+  linked_clone: true
+  ssh:
+    shell: /bin/sh
+
+provisioner:
+  init_environment: |
+    echo 'auto_accept: true' > /tmp/auto-accept-keys.conf
+    sudo mkdir -p /usr/local/etc/salt/master.d
+    sudo mv /tmp/auto-accept-keys.conf /usr/local/etc/salt/master.d/auto-accept-keys.conf
+    sudo pkg install -y shuf
+    sh -c 't=$(shuf -i 1-15 -n 1); echo Sleeping $t seconds; sleep $t'
+  sudo: true
+
+platforms:
+  - name: freebsd-122
+    driver:
+      box: bento/freebsd-12.2
+  - name: freebsd-114
+    driver:
+      box: bento/freebsd-11.4
+  - name: openbsd-68
+    driver:
+      box: generic/openbsd6
+      ssh:
+        shell: /bin/ksh
+    provisioner:
+      init_environment: |
+        echo 'auto_accept: true' > /tmp/auto-accept-keys.conf
+        sudo mkdir -p /etc/salt/master.d
+        sudo mv /tmp/auto-accept-keys.conf /etc/salt/master.d/auto-accept-keys.conf
+        sudo pkg_add coreutils
+        sh -c 't=$(gshuf -i 1-15 -n 1); echo Sleeping $t seconds; sleep $t'
+
+verifier:
+  command: pytest --cache-clear -v -s -ra --log-cli-level=debug -k "not test_ping" tests/integration/
```

Comment:
`test_ping` is actually working for local testing (on an Ubuntu 20.04 host).  It hangs until timeout in GitHub Actions, likely to be an issue with the macOS runner.  Haven't researched in great detail but these could be relevant:

* https://github.com/paramiko/paramiko/issues/721
* https://github.com/paramiko/paramiko/issues/735